internal/config/ociartifact/ociartifact.go

```diff
@@ -24,54 +28,135 @@ func New() Impl {
 
 // Artifact can be used to manage OCI artifacts.
 type Artifact struct {
-	// MountPath is the local path containing the artifact data.
-	MountPath string
+	// Data is the actual artifact content.
+	Data []byte
+}
+
+// PullOptions can be used to customize the pull behavior.
+type PullOptions struct {
+	// SystemContext is the context used for pull.
+	SystemContext *types.SystemContext
 
-	// Cleanup has to be called if the artifact is not used any more.
-	Cleanup func()
+	// EnforceConfigMediaType can be set to enforce a specific manifest config media type.
+	EnforceConfigMediaType string
+
+	// MaxSize is the maximum size of the artifact to be allowed to get pulled.
+	// Will be set to a default of 1MiB if not specified (zero) or below zero.
+	MaxSize int
 }
 
 // defaultImpl is the default implementation for the OCI artifact handling.
 type defaultImpl struct{}
 
-// Pull downloads and mounts the artifact content by using the provided ref.
-func (*defaultImpl) Pull(ctx context.Context, sys *types.SystemContext, ref string) (*Artifact, error) {
-	log.Infof(ctx, "Pulling OCI artifact from ref: %s", ref)
+// defaultMaxArtifactSize is the default maximum artifact size.
+const defaultMaxArtifactSize = 1024 * 1024 // 1 MiB
+
+// Pull downloads the artifact content by using the provided image name and the specified options.
+func (*defaultImpl) Pull(ctx context.Context, img string, opts *PullOptions) (*Artifact, error) {
+	log.Infof(ctx, "Pulling OCI artifact from ref: %s", img)
 
-	storeOpts, err := storage.DefaultStoreOptions(false, 0)
+	name, err := reference.ParseNormalizedNamed(img)
```

Comment:
(This requires the caller to explicitly include a tag or digest, it does not default to `:latest`. That may well be fine.)

Reply:
Fixed by using `reference.TagNameOnly(name)`.